packages/api/src/controllers/access-control.ts

```diff
@@ -40,83 +44,181 @@ app.post(
+          );
+        } else {
+          console.log(`
+            access-control: gate: content with playbackId=${playbackId} is gated but webhook=${webhook.id} returned status code ${statusCode}, disallowing playback
+          `);
+          throw new BadGatewayError(
+            "Content is gated and corresponding webhook failed"
+          );
+        }
+      default:
         throw new BadRequestError(
-          "Multiple signing keys found for the same public key."
+          `unknown playbackPolicy type: ${playbackPolicyType}`
         );
-      }
-
-      const signingKey = signingKeyOutput[0];
-
-      if (signingKey.userId !== content.userId) {
-        console.log(`
-          access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the content and the public key pub=${req.body.pub} do not share the same owner, signingKeyUserId=${signingKey.userId}, contentUserId=${content.userId}
-        `);
-        throw new NotFoundError("Content not found");
-      }
-
-      if (signingKey.disabled || signingKey.deleted) {
-        const signingKeyLog = JSON.stringify(JSON.stringify(signingKey));
-        console.log(`
-          access-control: gate: disallowing playback for contentId=${content.id} with playbackId=${playbackId} the public key pub=${signingKey.id} is disabled or deleted, signingKey=${signingKeyLog}
-        `);
-        throw new ForbiddenError("The public key is disabled or deleted");
-      }
-
-      tracking.recordSigningKeyValidation(signingKey.id);
-      res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
-      res.status(204);
-      return res.end();
-    } else {
-      console.log(`
-        access-control: gate: content with playbackId ${playbackId} is gated but playbackPolicyType ${playbackPolicyType} is not supported, disallowing playback
-      `);
-      throw new BadRequestError(
-        `unknown playbackPolicy type: ${playbackPolicyType}`
-      );
     }
   }
 );
 
+async function fireGateWebhook(
+  webhook: DBWebhook,
+  plabackPolicy: PlaybackPolicy,
+  accessKey: string
+) {
+  let timestamp = Date.now();
+  let params = {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+      "user-agent": "livepeer.studio",
+    },
+    timeout: WEBHOOK_TIMEOUT,
+    body: JSON.stringify({
+      context: plabackPolicy.webhookContext,
+      accessKey: accessKey,
+      timestamp: timestamp,
+    }),
+  };
+
+  const sigHeaders = signatureHeaders(
+    params.body,
+    webhook.sharedSecret,
+    timestamp
+  );
+  params.headers = { ...params.headers, ...sigHeaders };
+
+  const startTime = process.hrtime();
+  let resp: Response;
+  let errorMessage: string;
+  let statusCode: number;
+
+  try {
+    resp = await fetchWithTimeout(webhook.url, params);
+    // TODO: remember to handle redirects
+    statusCode = resp.status;
+
+    if (resp.status <= 200 || resp.status > 400) {
```

Comment:
From the log below I think you meant
```suggestion
    if (resp.status < 200 || resp.status >= 300) {
```